### ISSUE TITLE
debug build of shell.nix

### ISSUE THREAD

**Issue by InLaw** [cript0nauta/pynixify#46]
I have the following pkgs but the build process has some issues with it.
What is the best way to debug it
- e.g. currently i don't know where doit is in as a dependency

```
aiohttp 
#arcgis 
area 
asyncio 
autoflake 
#avro 
#black==19.10.b0
bokeh 
bottle 
chart-studio 
clang 
conda 
dash 
dash-leaflet 
dask 
#dbt 
#dephell 
descartes 
django 
earthengine-api 
#eli5 
fastapi 
Fiona 
flake8 
flask 
folium 
fuzzywuzzy
GeoAlchemy2 
geofeather 
geojsonio 
geopandas 
geopy 
geotable 
# gprof2dot  # doit issue
graphviz 
greedy 
h2o 
holoviews 
hunter 
hvplot 
imbalanced-learn 
ipdb 
ipyparallel 
ipython 
ipywidgets 
#jq 
#json2json 
#json2sql 
json2yaml 
jupyter_contrib_nbextensions 
jupyterhub 
jupyterlab 
jupyter_nbextensions_configurator 
keras 
lidar 
lime 
line-profiler 
Markdown 
missingno 
momepy 
more-itertools 
mypy 
networkx 
#Nikola 
objgraph 
# ogr2wkt  # issues
openpyxl 
orange3 
osmapi 
osm-diff-tool 
osm-graph 
osmnet 
OSMPythonTools 
overpass 
#overpy2 
pandana 
pandas 
panel 
parquet 
pip 
pip-tools 
pipx 
plop 
plotly 
poetry2conda 
pprint 
pprofile 
pyarrow 
pycsw 
pydal 
pygeos 
pygis 
pygments 
pyinstrument 
pymc3 
Pympler 
pyperf 
pyproj 
pyshp 
#py-spy 
pytest 
pywps 
quinn 
rasterio 
rasterstats 
recurse 
reportlab 
rio-cogeo 
rsgis 
scikit-learn 
seaborn 
shap 
Shapely 
simplejson 
spatial 
stackimpact 
#tensorflow 
tifffile 
torch 
tqdm 
tuna 
urbansim 
validjson 
vega 
verde 
VisiData 
voila 
vprof 
whitebox 
widgetsnbextension 
xarray 
xarray-spatial 
xdot 
xgboost 
XlsxWriter 
fuzzywuzzy
#spacy
nltk 
gensim
#word2vec 
quepy 
textblob 
Pattern 
PyNLPl 
corenlp
polyglot
```

currently it says:

```
builder for '/nix/store/f5m2r33m0794pbkcimjan3l0yp3x5jly-python3.7-doit-0.32.0.drv' failed with exit code 1
cannot build derivation '/nix/store/n3dd16sfdy7m46yzpgla3wn2gfm1nybd-python3-3.7.8-env.drv': 1 dependencies couldn't be built
error: --- Error --- nix-shell
build of '/nix/store/n3dd16sfdy7m46yzpgla3wn2gfm1nybd-python3-3.7.8-env.drv' failed
```

**Comment by cript0nauta** [comment]
The output of pynixify should be good enough for this usecase. For example, lets say you want to use Nikola, which has doit as a dependency. You will see something like this in pynixify's output:

```
Resolving doit>=0.30.1 (from NixPackage(attr=Nikola, version=8.0.4))
```

indicating that the doit requirement came from Nikola.

**Comment by cript0nauta** [comment]
Hi! Did the output of pynixify worked ok? Do you need anything else to fix this particular issue? Otherwise I'll close it.

**Comment by InLaw** [comment]
`pynixify.exceptions.NoMatchingVersionFound: New requirement docutils<0.16,>=0.10 (from NixPackage(attr=botocore, version=1.17.16)) does not match already installed docutils==0.16`

How can I find out due to which pkgs docutils is strictly fixed to  `docutils==0.16`

**Comment by cript0nauta** [comment]
When it says `already installed docutils==0.16`, it says the current version of docutils is 0.16. It's not that there was a strict `docutils==0.16` requirement. So before the exception, you'll probably see something like `Resolving docutils (from NixPackage(attr=sompackage, version=8.0.4))`.I can see the reason of confussion here. The error message could be improved to give more useful feedback.

I noted the strict requirement of an older version of docutils is fixed in botocore's definition: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/botocore/default.nix#L32. But just as with #45, pynixify isn't interpreting that.

As a workaround, you can put `docutils<0.16` as your first requirement in order to force the usage of an older PyPI version. docutils is pure python, so I think it will work ok.

**Comment by InLaw** [comment]
if I try to execute 

nix-shell shell.nix 
it results in:

```
builder for '/nix/store/6agg2fi4q26gc73qlqi9lxg1ac8531s4-python3.7-pandana-0.5.1.drv' failed with exit code 1
cannot build derivation '/nix/store/75b8715fy9xg5kr0aq7795d4bnlry5ks-python3-3.7.8-env.drv': 1 dependencies couldn't be built
error: --- Error --- nix-shell
build of '/nix/store/75b8715fy9xg5kr0aq7795d4bnlry5ks-python3-3.7.8-env.drv' failed
```


----
pandana fails because of 
`ModuleNotFoundError: No module named 'numpy'`
?

---
how can I now solve that issue?
